### ISSUE TITLE
[Feature request]: Option to prevent Flatpak app opening browser/other application

### ISSUE THREAD

**Issue by mariomadproductions** [flatpak/xdg-desktop-portal#708]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

Some programs (e.g. Discord) open your browser to a special URL in order to get your discord login info from the browser session. This is not a major privacy/security risk as this presumably requires the discord server to "cooperate", but it does seem wrong that an app can open arbitrary URLs without user permission (discord does this on startup without asking). I think it would be a good idea if this could be prevented somehow, using the flatpak per-app restrictions that users enable.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/flatpak/xdg-desktop-portal/issues/708#issuecomment-1035198636

@mariomadproductions, it shouldn't be the default, however. I can think of almost no people I know of who would be content with agreeing to view a URI every time one is invoked, myself included.

**Comment by mariomadproductions** [comment]
> > [#708 (comment)](https://github.com/flatpak/xdg-desktop-portal/issues/708#issuecomment-1035198636)
> 
> @mariomadproductions, it shouldn't be the default, however. I can think of almost no people I know of who would be content with agreeing to view a URI every time one is invoked, myself included.

Yeah probably shouldn't happen every time unless the user asks for that. One way could be to have a "don't ask again for this program" checkbox in the Yes/No pop-up (in addition to something changeable in the settings).
